### PR TITLE
smell issue#21- return authedRequest

### PR DESCRIPTION
**Pull request by ME-ON1**
As suggested by the smell issue [#21](https://sonarcloud.io/project/issues?id=SimplQ_simplQ-frontend&open=AXe6u6BMmUtjMyP-a6uy&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

### CODE REVIEW

File: simplq/src/store/asyncActions/getUserQueues.js [hunk 1]
```diff
@@ -19,9 +19,8 @@ const useGetUserQueues = () => {
   const makeAuthedRequest = useMakeAuthedRequest();
 
   const getUserQueues = createAsyncThunk(typePrefix, async () => {
-    const authedRequest = makeAuthedRequest(RequestFactory.getUserQueues());
-    const response = await authedRequest;
-    return response;
+    const authedRequest = await makeAuthedRequest(RequestFactory.getUserQueues());
+    return authedRequest;
```
Comment (daltonfury42) [inline simplq/src/store/asyncActions/getUserQueues.js:23]: Let's do this in one line:
```
return await makeAuthedRequest(RequestFactory.getUserQueues());
```

**Review by daltonfury42 — CHANGES_REQUESTED** [review]

File: simplq/src/store/asyncActions/getUserQueues.js (hunk 1 above)
Comment (ME-ON1) [reply simplq/src/store/asyncActions/getUserQueues.js:23]: Ohh right,  On it !!

Comment (ME-ON1) [reply simplq/src/store/asyncActions/getUserQueues.js]: Done please check that again !

**Comment by daltonfury42** [comment]
Looks like you missed an await

**Comment by ME-ON1** [comment]
NO, i removed that purposefully. 
`return await makeAuthedRequest(RequestFactory.getUserQueues());` 
wasn't possible as it was causing error by eLint  
`error  Redundant use of "await" on a return value  no-return-await` 
and after some research i found that 
`return await fun()` is same as `return  fun()` 
inside a async function.

**Review by daltonfury42 — APPROVED** [review]
Nice. Didn't know this.

**Comment by maaverik** [comment]
> and after some research i found that
> `return await fun()` is same as `return fun()`
> inside a async function.
Hi @ME-ON1, thanks for the PR. Just wanted to double-check if you sure that they are equivalent. This [MDN doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) at the end says that they are not equivalent and if so, I think we should retain await in the function because we are not awaiting anywhere outside the thunk. Could you please check again if this change would cause any problems?

**Comment by daltonfury42** [comment]
They are not equivalent in the sense that the function doesn't block. It immediately returns as a promise, and hence is little more performant.

The difference is very subtle, we would need to await only if say we want to block and catch errors inside the function etc. Since we don't want to do that, this should work for us.

**Pull request merged**